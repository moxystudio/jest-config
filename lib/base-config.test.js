'use strict';

jest.mock('ci-info', () => ({ isCI: false }));

process.getuid = () => 1;

const baseConfig = require('./base-config');

it('should match snapshot', () => {
    expect(baseConfig()).toMatchSnapshot();
});

it('should set test environment', () => {
    expect(baseConfig().testEnvironment).toBe('jsdom');
    expect(baseConfig('jsdom').testEnvironment).toBe('jsdom');
    expect(baseConfig('node').testEnvironment).toBe('node');
});
