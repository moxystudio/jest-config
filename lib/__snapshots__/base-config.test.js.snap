// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
Object {
  "automock": false,
  "bail": 0,
  "browser": false,
  "cache": true,
  "cacheDirectory": "<TEMP_DIR>/jest_1",
  "changedFilesWithAncestor": false,
  "clearMocks": false,
  "collectCoverage": false,
  "coveragePathIgnorePatterns": Array [
    "/node_modules/",
  ],
  "coverageProvider": "babel",
  "coverageReporters": Array [
    "json",
    "text",
    "lcov",
    "clover",
  ],
  "coverageThreshold": Object {
    "global": Object {
      "branches": 80,
      "functions": 80,
      "lines": 80,
      "statements": -10,
    },
  },
  "errorOnDeprecated": false,
  "expand": false,
  "forceCoverageMatch": Array [],
  "globals": Object {},
  "haste": Object {
    "computeSha1": false,
    "providesModuleNodeModules": Array [],
    "throwOnModuleCollision": false,
  },
  "maxConcurrency": 5,
  "maxWorkers": "50%",
  "moduleDirectories": Array [
    "node_modules",
  ],
  "moduleFileExtensions": Array [
    "js",
    "json",
    "jsx",
    "ts",
    "tsx",
    "node",
  ],
  "moduleNameMapper": Object {},
  "modulePathIgnorePatterns": Array [],
  "noStackTrace": false,
  "notify": false,
  "notifyMode": "failure-change",
  "prettierPath": "prettier",
  "resetMocks": false,
  "resetModules": false,
  "restoreMocks": false,
  "roots": Array [
    "<rootDir>",
  ],
  "runTestsByPath": false,
  "runner": "jest-runner",
  "setupFiles": Array [],
  "setupFilesAfterEnv": Array [],
  "skipFilter": false,
  "snapshotSerializers": Array [
    "<PROJECT_ROOT>/node_modules/jest-serializer-path/lib/index.js",
  ],
  "testEnvironment": "jsdom",
  "testEnvironmentOptions": Object {},
  "testFailureExitCode": 1,
  "testLocationInResults": false,
  "testMatch": Array [
    "**/?(*.)test.[jt]s",
  ],
  "testPathIgnorePatterns": Array [
    "/node_modules/",
    "/docusaurus/",
  ],
  "testRegex": Array [],
  "testRunner": "jasmine2",
  "testSequencer": "@jest/test-sequencer",
  "testURL": "http://localhost",
  "timers": "real",
  "transform": Object {
    "/.[jt]s$": "<PROJECT_ROOT>/node_modules/babel-jest/build/index.js",
  },
  "transformIgnorePatterns": Array [
    "/node_modules/",
  ],
  "useStderr": false,
  "watch": false,
  "watchPathIgnorePatterns": Array [],
  "watchman": true,
}
`;
